raiden/tests/integration/test_raidenservice.py

```diff
@@ -162,3 +163,42 @@ def num_matching_queued_messages(room: str, message_type: Type) -> int:
         user_deposit=app0.raiden.chain.user_deposit(user_deposit_address),
     )
     app0_restart.start()
+
+
+@pytest.mark.parametrize("start_raiden_apps", [False])
+@pytest.mark.parametrize("deposit", [0])
+@pytest.mark.parametrize("channels_per_node", [CHAIN])
+@pytest.mark.parametrize("number_of_nodes", [2])
+def test_alarm_task_first_run_syncs_blockchain_events(raiden_network):
+    """
+    Test that the alarm tasks syncs blockchain events at the end of its first run
+
+    Test for https://github.com/raiden-network/raiden/issues/4498
+    """
+    # These apps have had channels created but are not yet started
+    app0, _ = raiden_network
+
+    original_first_run = app0.raiden._prepare_and_execute_alarm_first_run
+
+    def first_run_with_check(from_block):
+        """
+        This function simply enhances the alarm task first run
+
+        The enhanced version has a check for channels being available right after
+        the first run of the alarm task
+        """
+        original_first_run(from_block)
```

Comment:
This has race conditions, since `app0.start` will do the complete app initialization, the alarm task will be executed, and the channels may get pickup because of the alarm task and not the `_prepare_and_execute_alarm_first_run`.

Another solution is to just not monkeypatch it, and call the initialization methods up to the `_prepare_and_execute_alarm_first_run`, then assert on the expected result. Since the alarm task is not initialized then we know for sure there are not races with it.

Reply:
Hmmm it is monkey patched. If it's not picked up by the first run of the alarm task this function should fail.

Reply:
> Hmmm it is monkey patched

`_prepare_and_execute_alarm_first_run` is, [`_start_alarm_task`](https://github.com/raiden-network/raiden/blob/1c1f5dba1a3aabdc87cbdfcb9c72ac0443f02aa2/raiden/raiden_service.py#L448) is not, and it will run, there is no synchronization among that method and the test, so there is a race condition.

Please avoid monkey patching.

Reply:
okay, you're right, the monkeypatched function will run before the alarm task is started.